packages/content-conformance/src/rules/must-have-h1.js

```diff
@@ -0,0 +1,32 @@
+// eslint-disable-next-line import/no-anonymous-default-export
+/** @type {import('../types.js').ConformanceRuleBase} */
+export default {
+  type: 'content',
+  id: 'must-have-h1',
+  description: 'Each content file must contain an H1 the top.',
+  executor: {
+    async contentFile(file, context) {
+      let hasH1 = false
+      let hasVisitedAHeading = false
+
+      file.visit(['heading'], (node) => {
+        hasVisitedAHeading = true
+
+        if (node.depth === 1) {
+          hasH1 = true
+        }
+
+        if (!hasH1 && hasVisitedAHeading) {
+          context.report(
+            'Must have a level 1 heading at the top of the file.',
+            file,
+            node
+          )
+          // stop traversing
+          return false
+        }
+      })
+    },
```

Comment:
So this will validate that an H1 exists before any other heading, but it doesn't necessarily validate that the H1 is the first node/element. Consider this case:

```
I am some text.

# I am an H1
``

This should fail, as the first node isn't an H1.

Reply:
Ah true. I will get on that